File: src/main/java/br/com/bookstore/service/BookService.java
package br.com.bookstore.service;

import br.com.bookstore.model.Book;
import br.com.bookstore.model.repository.BookRepository;

import java.util.List;

public class BookService {

    private final BookRepository bookRepository;

    public BookService(BookRepository bookRepository) {
        this.bookRepository = bookRepository;
    }

    public Book saveBook(Book book) {
        return bookRepository.save(book);
    }

    public Book updateBook(Book book) {
        if (!bookRepository.existsById(book.getId())) {
            throw new IllegalArgumentException("Book with id " + book.getId() + " not found");
        }
        return bookRepository.save(book);
    }

    public List<Book> listAllBooks() {
        return bookRepository.findAll();
    }

    public Book getBookById(Long id) {
        return bookRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("Book with id " + id + " not found"));
    }

    public void deleteBook(Long id) {
        if (!bookRepository.existsById(id)) {
            throw new IllegalArgumentException("Book with id " + id + " not found");
        }
        bookRepository.deleteById(id);
    }
}

File: src/main/java/br/com/bookstore/service/BookService.kt
package br.com.bookstore.service

import br.com.bookstore.model.Book
import br.com.bookstore.model.repository.BookRepository

class BookService(
    private val bookRepository: BookRepository
) {
    fun saveBook(book: Book): Book {
        return bookRepository.save(book)
    }

    fun updateBook(book: Book): Book {
        if (!bookRepository.existsById(book.id)) {
            throw IllegalArgumentException("Book with id \${book.id} not found")
        }
        return bookRepository.save(book)
    }

    fun listAllBooks(): List<Book> {
        return bookRepository.findAll()
    }

    fun getBookById(id: Long): Book {
        return bookRepository.findById(id)
            .orElseThrow { IllegalArgumentException("Book with id \$id not found") }
    }

    fun deleteBook(id: Long) {
        if (!bookRepository.existsById(id)) {
            throw IllegalArgumentException("Book with id \$id not found")
        }
        bookRepository.deleteById(id)
    }
}